src/systems/imu/Imu.cc

```diff
@@ -213,6 +214,19 @@ void ImuPrivate::AddSensor(
         data.ImuSensor()->OrientationEnabled());
   }
 
+  // Get world frame orientation and heading
+  double heading = 0.0;
+
+  ignition::gazebo::World world(worldEntity);
+  if (world.SphericalCoordinates(_ecm))
+  {
+    auto sphericalCoordinates = world.SphericalCoordinates(_ecm).value();
+    heading = sphericalCoordinates.HeadingOffset().Radian();
+  }
+
+  sensor->SetWorldFrameOrientation(math::Quaterniond(0, 0, heading),
```

Comment:
@iche033 will be making a release son, we were waiting for https://github.com/ignitionrobotics/ign-sensors/pull/192, which seems to be merged. 

Yes, the sensor should support both absolute and relative reference frames. Originally, the system sets the orientation as "absolute" i.e. wrt to the world, but there can be case when we have a "heading" in spherical coordinates, and that information needs to be supplied to the sensor, hence the ``SetWorldFrameOrientation`` call. I'll include a comment to explain.

Reply:
> Originally, the system sets the orientation as "absolute" i.e. wrt to the world

Hmm, does it? That `ImuSensor::SetOrientationReference()` call is setting the initial IMU body frame world pose as orientation reference, which is fine if one wanted to simulate orientation estimates for an IMU w/o a magnetometer.  

It may be worth adding a few more tests for when no world frame orientation is specified, to fully characterize sensor behavior in each case. WDYT @adityapande-1995?